### ISSUE TITLE
Not able to start meeting in 2.2.0-beta9

### ISSUE THREAD

**Issue by netlabs** [bigbluebutton/bigbluebutton#7316]
Hey, I have updated to 2.2.0-beta5 which is available at http://docs.bigbluebutton.org/2.2/install.html#bbb-installsh now I am not able to start meeting. I have tried restarting many times but didn't work. Please find the attached screenshots for reference errors. In BBB-status also all are fine.

![image](https://user-images.githubusercontent.com/45689540/56846045-96e78480-68e7-11e9-97a2-aaaca1b96348.png)
![image](https://user-images.githubusercontent.com/45689540/56846050-aa92eb00-68e7-11e9-9b02-b4b6c8d457dc.png)

**Comment by ffdixon** [comment]
Did you get any errors during the upgrade?  Did you do a `sudo bbb-conf --setip <IP/hostname>` after the upgrade?

Also, how are you logging in?  Can you try it with the API examples, see

 http://docs.bigbluebutton.org/2.2/install.html#5-install-api-demos-optional

Can you post the output of

  `sudo bbb-conf --check`

**Comment by netlabs** [comment]
No I didn't get any errors. No I haven't given this command sudo bbb-conf --setip <IP/hostname>.
I am trying to start meeting as usual as in 2.0. But when I click on start I get this blue screen with login error.

**Comment by ffdixon** [comment]
> No I haven't given this command sudo bbb-conf --setip <IP/hostname>.

You need to use `sudo bbb-conf --setip <IP/hostname>` to re-assign the hostname after upgrading the packages.   

Try that and let us know if your able to join a meeting after the command finishes.

**Comment by netlabs** [comment]
> 
> 
> > No I haven't given this command sudo bbb-conf --setip <IP/hostname>.
> 
> You need to use `sudo bbb-conf --setip <IP/hostname>` to re-assign the hostname after upgrading the packages.
> 
> Try that and let us know if your able to join a meeting after the command finishes.

Even after trying this I am getting same error

**Comment by ffdixon** [comment]
Can you update to BigBlueButton 2.2-beta-6, released earlier this week

  https://groups.google.com/d/msg/bigbluebutton-dev/mgmDVi8yFGU/9AJX9ZJ4AQAJ

and let us know if you get the same issue.  You should be able to login fin with 2.2-beta-5, but something isn't right in your setup.  Let's get you on the latest version and then start investigating further.

**Comment by netlabs** [comment]
> 
> 
> Can you update to BigBlueButton 2.2-beta-6, released earlier this week
> 
> https://groups.google.com/d/msg/bigbluebutton-dev/mgmDVi8yFGU/9AJX9ZJ4AQAJ
> 
> and let us know if you get the same issue. You should be able to login fin with 2.2-beta-5, but something isn't right in your setup. Let's get you on the latest version and then start investigating further.

Still same issue


![image](https://user-images.githubusercontent.com/45689540/57130011-d580b180-6db5-11e9-8946-69d6809b8ad6.png)

**Comment by ffdixon** [comment]
Hi,

When we try accessing the server we're getting a Welcome to nginx, which suggests that nginx isn't properly configured to route the incoming request.  Can you show the full output for `sudo bbb-conf --check`

**Comment by netlabs** [comment]
> 
> 
> Hi,
> 
> When we try accessing the server we're getting a Welcome to nginx, which suggests that nginx isn't properly configured to route the incoming request. Can you show the full output for `sudo bbb-conf --check`

Use https for accessing page.
![image](https://user-images.githubusercontent.com/45689540/57173665-285b7700-6e51-11e9-829f-3f9a1f2d1748.png)
![image](https://user-images.githubusercontent.com/45689540/57173667-33aea280-6e51-11e9-9ca3-70c7514e82fe.png)

**Comment by ffdixon** [comment]
It looks like your server has in internal IP address 172.21.10.14 and is behind a firewall at 27.34.244.186.  See

  http://docs.bigbluebutton.org/2.2/configure-firewall.html

**Comment by netlabs** [comment]
I have updated to beta 9 but still same issue. I have checked the document with 2.2 firewall and others too. But no resolution. Please help me.

**Comment by antobinary** [comment]
Hi @netlabs 
In one of your screenshots I see url `.../html5client/users`. We moved away from using `users` in 2.2 so I would bet that [at least some part of] your html5 client is still on BBB 2.0. Did you perhaps bookmark /users? This is what you should be seeing (and using) now:
https://test.bigbluebutton.org/html5client/join?sessionToken=xxxxxxxxx

![Screenshot from 2019-05-30 08-02-39](https://user-images.githubusercontent.com/6312397/58631700-619cd080-82b1-11e9-9b0d-d84555d0647d.png)

**Comment by netlabs** [comment]
@antobinary
That might be the possible error. But the link you have given me shows invalid session id. I don't get that error message.

So if you're assumption is correct. How should I proceed further?

**Comment by antobinary** [comment]
@netlabs 
This is not related to 7502 or 7446. The issue here is that GreenLight redirects you to an incorrect URL.

If you were to apt install the `bbb-demo` package you should be able to log in fine via HTML5 client through `<yourserver>/demo/demoHTML5.jsp`
Once you confirm that you are able to log in fine, you can `apt purge bbb-demo`
At that point we will know with 100% certainty that what needs to be addressed is the GreenLight bit.

I do not have much experience with GL but from what I grasp the redirection url is part of the env and can be configured by following http://docs.bigbluebutton.org/greenlight/gl-install.html#3-configure-greenlight
In short, 2.0 redirected you to `<yourserver>/html5client/users`
In 2.2 we require `sessionToken` and the expected url is of the form `<yourserver>/html5client/join?sessionToken=<some session token>`

**Comment by capilkey** [comment]
The client URL is dictated by bbb-web and it's the same for both 2.2 and 2.0. What was changed was how the HTML5 client itself redirects once the /html5client/join is reached. To me it looks like either the client has been customized or it isn't the beta-9 version of the package. The /html5client/users endpoint and the look of the error page are the giveaways.

The 2.2-beta9 error page looks like this:
![image](https://user-images.githubusercontent.com/1395090/58646804-2d86d700-82d4-11e9-9f4d-fe09b7691c6f.png)

**Comment by netlabs** [comment]
> 
> 
> @netlabs
> This is not related to 7502 or 7446. The issue here is that GreenLight redirects you to an incorrect URL.
> 
> If you were to apt install the `bbb-demo` package you should be able to log in fine via HTML5 client through `<yourserver>/demo/demoHTML5.jsp`
> Once you confirm that you are able to log in fine, you can `apt purge bbb-demo`
> At that point we will know with 100% certainty that what needs to be addressed is the GreenLight bit.
> 
> I do not have much experience with GL but from what I grasp the redirection url is part of the env and can be configured by following http://docs.bigbluebutton.org/greenlight/gl-install.html#3-configure-greenlight
> In short, 2.0 redirected you to `<yourserver>/html5client/users`
> In 2.2 we require `sessionToken` and the expected url is of the form `<yourserver>/html5client/join?sessionToken=<some session token>`

 I have taken a new greenlight env file from git and checked. After doing this also facing the same error.

**Comment by capilkey** [comment]
@netlabs What about the questions about the client you're running? It definitely doesn't look or function like the current 2.2 client.

**Comment by netlabs** [comment]
> 
> 
> @netlabs What about the questions about the client you're running? It definitely doesn't look or function like the current 2.2 client.

![image](https://user-images.githubusercontent.com/45689540/58682598-dc87e900-838e-11e9-9bcb-e66152fe1c3e.png)


![image](https://user-images.githubusercontent.com/45689540/58682605-e578ba80-838e-11e9-8968-8b519d4b2b33.png)

**Comment by capilkey** [comment]
Those images don't show enough. It would be "bbb-conf --version" and also whether or not you're running a custom client. The HTML5 code that is running on your server is out of date and incompatible with the other components on the server.

**Comment by netlabs** [comment]
> 
> 
> Those images don't show enough. It would be "bbb-conf --version" and also whether or not you're running a custom build of the HTML5 code. The HTML5 code that is running on your server is out of date and incompatible with the other components on the server.

Please check this.

![image](https://user-images.githubusercontent.com/45689540/58683026-9a5fa700-8390-11e9-8689-a170ad5176e2.png)

**Comment by capilkey** [comment]
The html5 package is still 2.0.

What about "whereis yq". If that shows that it isn't found then you need to:
```
add-apt-repository ppa:rmescandon/yq -y
apt-get update
apt dist-upgrade
```

Another option is to purge the HTML5 package and then reinstall it.
```
apt-get purge bbb-html5
apt-get install bigbluebutton
```

The html5 package version should be `ii  bbb-html5     1:2.2.0-497`

**Comment by netlabs** [comment]
> 
> 
> The html5 package is still 2.0.
> 
> What about "whereis yq". If that shows that it isn't found then you need to:
> 
> ```
> add-apt-repository ppa:rmescandon/yq -y
> apt-get update
> apt-get dist-upgrade
> ```
> 
> Another option is to purge the HTML5 package and then reinstall it.
> 
> ```
> apt-get purge bbb-html5
> apt-get install bigbluebutton
> ```
> 
> The html5 package version should be `ii bbb-html5 1:2.2.0-497`

Thanks a lot its working now.

**Comment by capilkey** [comment]
Glad to hear that you got it working
